clustered/client/src/main/java/org/ehcache/clustered/client/internal/store/operations/codecs/OperationsCodec.java

```diff
@@ -0,0 +1,58 @@
+/*
+ * Copyright Terracotta, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.ehcache.clustered.client.internal.store.operations.codecs;
+
+import org.ehcache.clustered.client.internal.store.operations.BaseOperation;
+import org.ehcache.clustered.client.internal.store.operations.PutOperation;
+
+import java.nio.ByteBuffer;
+
+public class OperationsCodec<K, V> {
+
+  public static final int BYTE_SIZE_BYTES = 1;
+  public static final int INT_SIZE_BYTES = 4;
+  public static final int LONG_SIZE_BYTES = 8;
+
+  private final OperationCodecFactory<K, V> codecFactory;
+
+  public OperationsCodec(final OperationCodecFactory<K, V> codecFactory) {
+    this.codecFactory = codecFactory;
+  }
+
+  public ByteBuffer encode(BaseOperation<K> operation) {
+    BaseOperation.Code opCode = operation.getOpCode();
+    switch (opCode) {
```

Comment:
Scratch that - `OperationCodecFactory` could be based on the operation concrete type, using a `Map<Class<? extends BaseOperation>,? extends OperationCodec>`